q2_qemistree/_hierarchy.py

```diff
@@ -24,12 +24,14 @@ def build_tree(relabeled_fingerprints: pd.DataFrame) -> TreeNode:
     This function makes a tree of relatedness between mass-spectrometry
     features using molecular substructure fingerprints.
     '''
+    leaves_ids = relabeled_fingerprints.index.tolist()
+    relabeled_fingerprints = relabeled_fingerprints.values
     distmat = pairwise_distances(X=relabeled_fingerprints,
```

Comment:
I don't think we are using relabeled_fingerprints anywhere else so it should be fine to take the values attribute directly, for example as:

```suggestion
    distmat = pairwise_distances(X=relabeled_fingerprints.values,
```

Reply:
great suggestion, I added this. I removed `leaves_ids`;  doing this on the fly too.